pkg/ipamd/ipamd.go

```diff
@@ -1750,6 +1788,26 @@ func (c *IPAMContext) isDatastorePoolTooHigh() bool {
 		return over > 0
 	}
 
+	//For the existing ENIs check if we can cleanup prefixes
+	if c.warmPrefixTargetDefined() {
+		total, used, _ := c.dataStore.GetStats()
+		available := total - used
+		_, maxIpsPerPrefix, _ := datastore.GetPrefixDelegationDefaults()
+		poolTooHigh := available > (maxIpsPerPrefix * (c.warmPrefixTarget+1))
+		if poolTooHigh {
+			logPoolStats(total, used, c.maxIPsPerENI, c.enableIpv4PrefixDelegation)
+			log.Debugf("Prefix pool is high: available (%d) > Warm prefix target (%d)+1 * maxIpsPerPrefix (%d)", available, c.warmPrefixTarget, maxIpsPerPrefix)
+		}
+		return poolTooHigh
+	}
 	// We only ever report the pool being too high if WARM_IP_TARGET or WARM_PREFIX_TARGET is set
 	return false
 }
+
+func (c *IPAMContext) warmPrefixTargetDefined() bool {
+	return c.warmPrefixTarget >= noWarmPrefixTarget && c.enableIpv4PrefixDelegation
+}
+
+func ceil(x, y int) int {
```

Comment:
I know why we called it this, but it is not at all like `Math.Ceil` and we should rename this function :P

Maybe
```suggestion
func divCeil(x, y int) int {
```

Reply:
Right, this is an alternate version to avoid floats :), will update it :)